diff --git a/Makefile b/Makefile
index 34b050ef..eaed7a28 100644
--- a/Makefile
+++ b/Makefile
@@ -541,6 +541,23 @@ else ifeq ($(platform), miyoo)
 	USE_CYCLONE := 1
 	USE_DRZ80 := 1
 
+# TrimUI Model S
+else ifeq ($(platform), trimui)
+	TARGET = $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	LDFLAGS += -shared -Wl,--version-script=link.T -Wl,-no-undefined -fno-PIC -flto
+	CFLAGS += -std=c99 -fomit-frame-pointer -ffast-math -mcpu=arm926ej-s -mtune=arm926ej-s -fno-PIC -flto
+	CXXFLAGS = $(CFLAGS) -fno-rtti -fno-exceptions
+	LIBS := -lc -lgcc -lm
+	CPU_ARCH := arm
+	ARM = 1
+	OPTIMIZE := -Ofast -DNDEBUG
+	ifeq (,$(DEBUG))
+		LDFLAGS += -s
+	endif
+
 # Emscripten
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_$(platform).bc
@@ -847,10 +864,19 @@ CFLAGS += -DRETRO_PROFILE=$(RETRO_PROFILE)
 # Disable optimization when debugging #####################
 ifeq ($(DEBUG), 1)
 	CFLAGS += -O0 -g3
+else ifneq (,$(OPTIMIZE))
+	CFLAGS += $(OPTIMIZE)
 else
 	CFLAGS += -O2 -DNDEBUG
 endif
 
+ifeq ($(PROFILE), GENERATE)
+	CFLAGS  += -fprofile-generate=./profile/mame2003_plus
+	LDFLAGS += -lgcov
+else ifeq ($(PROFILE), APPLY)
+	CFLAGS  += -fprofile-use -fprofile-dir=../profile/mame2003_plus -fbranch-probabilities
+endif
+
 ifneq (,$(findstring msvc,$(platform)))
 ifeq ($(DEBUG),1)
 	CFLAGS += -MTd
