diff --git a/Makefile b/Makefile
index 25ea7bc..7d923ad 100644
--- a/Makefile
+++ b/Makefile
@@ -39,6 +39,7 @@ ifeq ($(platform), unix)
 	ifneq ($(findstring Haiku,$(shell uname -a)),)
 		LIBM :=
 	endif
+	CFLAGS += -D_GNU_SOURCE
 
 # OS X
 else ifeq ($(platform), osx)
@@ -308,6 +309,21 @@ else ifeq ($(platform), miyoo)
 	CFLAGS += -O2 -mcpu=arm926ej-s
 	CXXFLAGS += -O2 -mcpu=arm926ej-s
 
+# TrimUI Model S
+else ifeq ($(platform), trimui)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--version-script=link.T
+	CFLAGS += -fomit-frame-pointer -march=armv5te -mtune=arm926ej-s -ffast-math -fno-PIC -flto -D_GNU_SOURCE
+	CXXFLAGS += -fomit-frame-pointer -march=armv5te -mtune=arm926ej-s -ffast-math -fno-PIC -flto -D_GNU_SOURCE
+	LDFLAGS += -flto
+	OPTIMIZE := -Ofast -DNDEBUG
+	ifeq (,$(DEBUG))
+		LDFLAGS += -s
+	endif
+
 else ifeq ($(platform), emscripten)
 	TARGET := $(TARGET_NAME)_libretro_$(platform).bc
 	STATIC_LINKING = 1
@@ -554,6 +570,17 @@ CXXFLAGS += -Wall $(INCFLAGS) $(INCFLAGS_PLATFORM)
 
 ifeq ($(DEBUG),1)
 CFLAGS += -O0 -g
+else ifneq (,$(OPTIMIZE))
+CFLAGS += $(OPTIMIZE)
+else
+CFLAGS += -O2 -DNDEBUG
+endif
+
+ifeq ($(PROFILE), GENERATE)
+CFLAGS  += -fprofile-generate=./profile/$(TARGET_NAME)
+LDFLAGS += -lgcov
+else ifeq ($(PROFILE), APPLY)
+CFLAGS  += -fprofile-use -fprofile-dir=../profile/$(TARGET_NAME) -fbranch-probabilities
 endif
 
 ifneq (,$(findstring msvc,$(platform)))
