diff --git a/libretro/libretro.c b/libretro/libretro.c
index 8069103..576bb10 100644
--- a/libretro/libretro.c
+++ b/libretro/libretro.c
@@ -1221,7 +1221,7 @@ void retro_run(void)
                         1 : 0;
 
             if (!skip_next_frame ||
-                (frameskip_counter >= FRAMESKIP_MAX))
+                (frameskip_counter >= frameskip_interval))
             {
                skip_next_frame   = 0;
                frameskip_counter = 0;
@@ -1238,7 +1238,7 @@ void retro_run(void)
                               1 : 0;
 
             if (!skip_next_frame ||
-                (frameskip_counter >= FRAMESKIP_MAX))
+                (frameskip_counter >= frameskip_interval))
             {
                skip_next_frame   = 0;
                frameskip_counter = 0;
