diff --git a/Makefile.libretro b/Makefile.libretro
index 9660e99..c69f854 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -496,6 +496,23 @@ else ifeq ($(platform), gcw0)
 	CFLAGS += -ffast-math -fomit-frame-pointer -march=mips32 -mtune=mips32r2 -mhard-float
 	CFLAGS += -DDINGUX
 
+# TrimUI Model S
+else ifeq ($(platform), trimui)
+	TARGET := $(TARGET_NAME)_libretro.so
+	CC = $(CROSS_COMPILE)gcc
+	CXX = $(CROSS_COMPILE)g++
+	AR = $(CROSS_COMPILE)ar
+	SHARED := -shared -Wl,--version-script=libretro/link.T
+	fpic := -fno-PIC
+	CFLAGS += $(PTHREAD_FLAGS)
+	CFLAGS += -ffast-math -fomit-frame-pointer -march=armv5te -mtune=arm926ej-s -flto -D_GNU_SOURCE
+	CXXFLAGS += -ffast-math -fomit-frame-pointer -march=armv5te -mtune=arm926ej-s -flto -D_GNU_SOURCE
+	LDFLAGS += -flto
+	OPTIMIZE := -Ofast -DNDEBUG
+	ifeq (,$(DEBUG))
+		LDFLAGS += -s
+	endif
+
 # RETROFW
 else ifeq ($(platform), retrofw)
 	TARGET := $(TARGET_NAME)_libretro.so
@@ -650,10 +667,19 @@ endif
 
 ifeq ($(DEBUG), 1)
 	CFLAGS += -O0 -g
+else ifneq (,$(OPTIMIZE))
+	CFLAGS += $(OPTIMIZE)
 else
 	CFLAGS += -O2 -DNDEBUG -DGIT_VERSION=\"$(GIT_VERSION)\"
 endif
 
+ifeq ($(PROFILE), GENERATE)
+	CFLAGS  += -fprofile-generate=./profile/$(TARGET_NAME)
+	LDFLAGS += -lgcov
+else ifeq ($(PROFILE), APPLY)
+	CFLAGS  += -fprofile-use -fprofile-dir=../profile/$(TARGET_NAME) -fbranch-probabilities
+endif
+
 include $(BUILD_DIR)/Makefile.common
 
 OBJECTS := $(SOURCES_C:.c=.o)
