diff --git a/libgambatte/libretro/libretro.cpp b/libgambatte/libretro/libretro.cpp
index 1c4471c..522054f 100644
--- a/libgambatte/libretro/libretro.cpp
+++ b/libgambatte/libretro/libretro.cpp
@@ -666,7 +666,8 @@ enum frame_blend_method
    FRAME_BLEND_NONE = 0,
    FRAME_BLEND_MIX,
    FRAME_BLEND_LCD_GHOSTING,
-   FRAME_BLEND_LCD_GHOSTING_FAST
+   FRAME_BLEND_LCD_GHOSTING_FAST,
+   FRAME_BLEND_LCD_GHOSTING_FASTEST
 };
 
 static enum frame_blend_method frame_blend_type  = FRAME_BLEND_NONE;
@@ -720,6 +721,41 @@ static void blend_frames_mix(void)
    }
 }
 
+/* Same as blend_frames_mix, but blending already blended frames to
+ * simulate lcd response */
+static void blend_frames_lcd_ghost_fastest(void)
+{
+   gambatte::video_pixel_t *curr = video_buf;
+   gambatte::video_pixel_t *prev = video_buf_prev_1;
+   size_t x, y;
+
+   for (y = 0; y < VIDEO_HEIGHT; y++)
+   {
+      for (x = 0; x < VIDEO_WIDTH; x++)
+      {
+         /* Get colours from current + previous frames */
+         gambatte::video_pixel_t rgb_curr = *(curr + x);
+         gambatte::video_pixel_t rgb_prev = *(prev + x);
+
+         /* Mix colours
+          * > "Mixing Packed RGB Pixels Efficiently"
+          *   http://blargg.8bitalley.com/info/rgb_mixing.html */
+#ifdef VIDEO_RGB565
+         *(curr + x) = (rgb_curr + rgb_prev + ((rgb_curr ^ rgb_prev) & 0x821)) >> 1;
+#elif defined(VIDEO_ABGR1555)
+         *(curr + x) = (rgb_curr + rgb_prev + ((rgb_curr ^ rgb_prev) & 0x521)) >> 1;
+#else
+         *(curr + x) = (rgb_curr + rgb_prev + ((rgb_curr ^ rgb_prev) & 0x10101)) >> 1;
+#endif
+         /* Store colours for next frame */
+         *(prev + x) = rgb_prev == *(curr + x) ? rgb_curr : *(curr + x);
+      }
+
+      curr += VIDEO_PITCH;
+      prev += VIDEO_PITCH;
+   }
+}
+
 static void blend_frames_lcd_ghost(void)
 {
    gambatte::video_pixel_t *curr   = video_buf;
@@ -994,6 +1030,11 @@ static void init_frame_blending(void)
          if (!allocate_video_buf_acc())
             return;
          break;
+      case FRAME_BLEND_LCD_GHOSTING_FASTEST:
+         /* Fastest ghosting requires a single accumulator buffer */
+         if (!allocate_video_buf_prev(&video_buf_prev_1))
+            return;
+         break;
       case FRAME_BLEND_NONE:
       default:
          /* Error condition - cannot happen
@@ -1037,6 +1078,9 @@ static void init_frame_blending(void)
       case FRAME_BLEND_LCD_GHOSTING_FAST:
          blend_frames = blend_frames_lcd_ghost_fast;
          return;
+      case FRAME_BLEND_LCD_GHOSTING_FASTEST:
+         blend_frames = blend_frames_lcd_ghost_fastest;
+         return;
       case FRAME_BLEND_NONE:
       default:
          /* Error condition - cannot happen
@@ -1111,6 +1155,8 @@ static void check_frame_blend_variable(void)
          frame_blend_type = FRAME_BLEND_LCD_GHOSTING;
       else if (!strcmp(var.value, "lcd_ghosting_fast"))
          frame_blend_type = FRAME_BLEND_LCD_GHOSTING_FAST;
+      else if (!strcmp(var.value, "lcd_ghosting_fastest"))
+         frame_blend_type = FRAME_BLEND_LCD_GHOSTING_FASTEST;
    }
 
    if (frame_blend_type == FRAME_BLEND_NONE)
diff --git a/libgambatte/libretro/libretro_core_options.h b/libgambatte/libretro/libretro_core_options.h
index 84213f7..5fa72e7 100644
--- a/libgambatte/libretro/libretro_core_options.h
+++ b/libgambatte/libretro/libretro_core_options.h
@@ -612,10 +612,11 @@ struct retro_core_option_v2_definition option_defs_us[] = {
       NULL,
       NULL,
       {
-         { "disabled",          NULL },
-         { "mix",               "Simple" },
-         { "lcd_ghosting",      "LCD Ghosting (Accurate)" },
-         { "lcd_ghosting_fast", "LCD Ghosting (Fast)" },
+         { "disabled",             NULL },
+         { "mix",                  "Simple" },
+         { "lcd_ghosting",         "LCD Ghosting (Accurate)" },
+         { "lcd_ghosting_fast",    "LCD Ghosting (Fast)" },
+         { "lcd_ghosting_fastest", "LCD Ghosting (Fastest)" },
          { NULL, NULL },
       },
       "disabled"
