diff --git a/Makefile.libretro b/Makefile.libretro
index 1842538..24cc029 100644
--- a/Makefile.libretro
+++ b/Makefile.libretro
@@ -386,6 +386,21 @@ else ifeq ($(platform), gcw0)
    CFLAGS += -fomit-frame-pointer -ffast-math -march=mips32 -mtune=mips32r2 -mhard-float
    CXXFLAGS += $(CFLAGS)
 
+# Trimui
+else ifeq ($(platform), trimui)
+   TARGET := $(TARGET_NAME)_libretro.so
+   CC = $(CROSS_COMPILE)gcc
+   CXX = $(CROSS_COMPILE)g++
+   AR = $(CROSS_COMPILE)ar
+   SHARED := -shared -Wl,--version-script=$(version_script)
+   LDFLAGS += -fno-PIC -flto
+   PLATFORM_DEFINES := -DCC_RESAMPLER -DCC_RESAMPLER_NO_HIGHPASS
+   CFLAGS += -fomit-frame-pointer -ffast-math -mcpu=arm926ej-s -mtune=arm926ej-s -fno-PIC -flto
+   OPTIMIZE := -Ofast -DNDEBUG
+   ifeq (,$(DEBUG))
+      LDFLAGS += -s
+   endif
+
 # RETROFW
 else ifeq ($(platform), retrofw)
    TARGET := $(TARGET_NAME)_libretro.so
@@ -635,25 +650,37 @@ else
 endif
 
 ifeq ($(DEBUG), 1)
-ifneq (,$(findstring msvc,$(platform)))
-   CFLAGS += -MTd
-   CXXFLAGS += -MTd
-   LDFLAGS += -DEBUG
-	CFLAGS += -Od -Zi -D_DEBUG
-	CXXFLAGS += -Od -Zi -D_DEBUG
-else
-	CFLAGS += -O0 -g
-	CXXFLAGS += -O0 -g
-endif
+   ifneq (,$(findstring msvc,$(platform)))
+      CFLAGS += -MTd
+      CXXFLAGS += -MTd
+      LDFLAGS += -DEBUG
+      CFLAGS += -Od -Zi -D_DEBUG
+      CXXFLAGS += -Od -Zi -D_DEBUG
+   else
+      CFLAGS += -O0 -g
+      CXXFLAGS += -O0 -g
+   endif
    CFLAGS   += -DDEBUG
    CXXFLAGS += -DDEBUG
 else
-ifneq (,$(findstring msvc,$(platform)))
-   CFLAGS   += -MT
-   CXXFLAGS += -MT
+   ifneq (,$(findstring msvc,$(platform)))
+      CFLAGS   += -MT
+      CXXFLAGS += -MT
+   endif
+   ifneq (,$(OPTIMIZE))
+      CFLAGS += $(OPTIMIZE)
+      CXXFLAGS += $(OPTIMIZE)
+   else
+      CFLAGS   += -O2 -DNDEBUG
+      CXXFLAGS += -O2 -DNDEBUG
+   endif
 endif
-   CFLAGS   += -O2 -DNDEBUG
-   CXXFLAGS += -O2 -DNDEBUG
+
+ifeq ($(PROFILE), GENERATE)
+   CFLAGS	+= -fprofile-generate=./profile/gambatte
+   LDFLAGS	+= -lgcov
+else ifeq ($(PROFILE), APPLY)
+   CFLAGS	+= -fprofile-use -fprofile-dir=../profile/gambatte -fbranch-probabilities
 endif
 
 ifeq (,$(findstring msvc,$(platform)))
